From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: mm: remove check_move_unevictable_pages()
Date: Wed, 21 Jun 2023 17:45:50 +0100

All callers have now been converted to call
check_move_unevictable_folios().

Link: https://lkml.kernel.org/r/20230621164557.3510324-7-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/swap.h |    1 -
 mm/vmscan.c          |   17 -----------------
 2 files changed, 18 deletions(-)

--- a/include/linux/swap.h~mm-remove-check_move_unevictable_pages
+++ a/include/linux/swap.h
@@ -439,7 +439,6 @@ static inline bool node_reclaim_enabled(
 }
 
 void check_move_unevictable_folios(struct folio_batch *fbatch);
-void check_move_unevictable_pages(struct pagevec *pvec);
 
 extern void __meminit kswapd_run(int nid);
 extern void __meminit kswapd_stop(int nid);
--- a/mm/vmscan.c~mm-remove-check_move_unevictable_pages
+++ a/mm/vmscan.c
@@ -8075,23 +8075,6 @@ int node_reclaim(struct pglist_data *pgd
 }
 #endif
 
-void check_move_unevictable_pages(struct pagevec *pvec)
-{
-	struct folio_batch fbatch;
-	unsigned i;
-
-	folio_batch_init(&fbatch);
-	for (i = 0; i < pvec->nr; i++) {
-		struct page *page = pvec->pages[i];
-
-		if (PageTransTail(page))
-			continue;
-		folio_batch_add(&fbatch, page_folio(page));
-	}
-	check_move_unevictable_folios(&fbatch);
-}
-EXPORT_SYMBOL_GPL(check_move_unevictable_pages);
-
 /**
  * check_move_unevictable_folios - Move evictable folios to appropriate zone
  * lru list
_
